src/App.jsx

```diff
@@ -1,21 +1,40 @@
-
-import './App.css'
-import { Client } from 'appwrite'; 
+import React, { useState, useEffect } from "react";
+import { useDispatch } from "react-redux";
+import "./App.css";
+import { login, logout } from "./store/authSlice";
+import authService from "./appwrite/auth";
+import { Header, Footer } from "../src/components/index";
 
 function App() {
-console.log(import.meta.env.VITE_APPWRITE_URL)
-const client = new Client();
-
-client
-    .setEndpoint('https://cloud.appwrite.io/v1')
-    .setProject('65e5c4c6b5cfc4dda432');
+  const [loading, setLoading] = useState(true);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    authService
+      .getCurrentUser()
+      .then((userData) => {
+        if (userData) {
+          dispatch(login({ userData }));
+        } else {
+          dispatch(logout());
+        }
+      })
+      .catch((err) => console.log(err))
+      .finally(() => setLoading(false));
+  }, []);
 
-  return (
-    <>
-   
-   <div>Hello React</div>
-    </>
-  )
+  return !loading ? (
```

Comment:
try to create a custom hook or function for API call

Reply:
created custome hook